test/msw-api/setup-worker/scenarios/iframe/iframe.test.ts

```diff
@@ -0,0 +1,70 @@
+import * as path from 'path'
+import { Frame } from 'playwright'
+import { pageWith } from 'page-with'
+
+declare namespace window {
+  export const request: () => Promise<any>
+}
+
+function findFrame(frame: Frame) {
+  return frame.name() === ''
+}
+
+test('intercepts a request from an iframe (nested client)', async () => {
+  const { page } = await pageWith({
+    example: path.resolve(__dirname, 'iframe.mocks.ts'),
+    markup: path.resolve(__dirname, 'iframe.page.html'),
+    contentBase: path.resolve(__dirname),
+  })
+
+  const frame = page.mainFrame().childFrames().find(findFrame)
+  await frame.evaluate(() => window.request())
+
+  const firstNameElement = await frame.waitForSelector('#first-name')
+  const firstName = await firstNameElement.evaluate((node) => node.textContent)
+
+  expect(firstName).toBe('John')
+})
+
+test('intercepts a request from a deeply nested iframe', async () => {
+  const { page } = await pageWith({
+    example: path.resolve(__dirname, 'iframe.mocks.ts'),
+    markup: path.resolve(__dirname, 'iframe-deep.page.html'),
```

Comment:
😱😱 I find this naming slightly headache-inducing.. what about following?
* page-without-iframe.html
* page-with-one-iframe.html
* page-with-two-nested-iframes.html
(leaving space for more levels in the future if needed and/or for page-with-two-parallel-iframes if you will need to test multiple iframes side by side)

Reply:
Good point about the naming. It's a little cryptic, I agree. The challenge comes from the way the in-browser tests are constructed: the compiled example code is always appended to the main page. That's why in the deeply nested iframe test suite I create an empty page and then load what is effectively the first testing scenario (`top -> nested iframe`) in _another_ iframe.

Reply:
I've renamed them to:

- `app.html` where the request logic lives.
- `page-in-iframe` where the's an iframe for the `app.html`.
- `page-in-nested-iframe` where the `page-in-iframe` is nested in another iframe.